analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,269 @@
+  unique() %>%
+  mutate(note=paste0("Found in atleast ",numSample, " samples in a group")) %>%
+  as.data.frame()
+
+
+```
+
+
+
+```{r}
+#filter QCGeneFiltered_filtFusion to keep recurrent fusions from above sample.count and fusion_calls.summary
+
+QCGeneFiltered_recFusion<-fusion_calls %>%
+  dplyr::filter(FusionName %in% unique(c(sample.count$FusionName,fusion_calls.summary$FusionName)))
+
+```
+
+
+
+
+
+```{r}
+# remove fusions that are in > numGroup
+group.count <- fusion_calls %>%
+  dplyr::select(FusionName, !!as.name(group)) %>%
+  unique() %>%
+  group_by(FusionName) %>%
+  dplyr::mutate(group.ct = n(),Sample = toString(!!(as.name(group)))) %>%
+  dplyr::filter(group.ct >numGroup)
+
+
+# remove multi-fused genes
+fusion_recurrent5_per_sample <- fusion_calls  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample, FusionName, Gene1A, Gene1B, Gene2A, Gene2B) %>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = GeneSymbol) %>%
+  # Remove columns without gene symbols
+  dplyr::filter(GeneSymbol != "") %>%
+  dplyr::arrange(Sample, FusionName) %>%
+  # Retain only distinct rows
+  dplyr::distinct() %>%
+  group_by(Sample,GeneSymbol) %>% 
+  dplyr::summarise(Gene.ct = n()) %>% 
+  dplyr::filter(Gene.ct>limitMultiFused) %>%
+  mutate(note=paste0("multfused " ,limitMultiFused, " times per sample"))
+
+
+
+```
+
+
+```{r}
+
+
+outputfolder<-"results"
+setwd(outputfolder)
+# filter QCGeneFiltered_recFusion to remove fusions found in more than 1 group 
+
+QCGeneFiltered_recFusionUniq<-QCGeneFiltered_recFusion %>%
+  dplyr::filter(!FusionName %in% group.count$FusionName) %>%
+  dplyr::filter(!Gene1A %in% fusion_recurrent5_per_sample$GeneSymbol |
+                  !Gene2A %in% fusion_recurrent5_per_sample$GeneSymbol |
+                  !Gene1B %in% fusion_recurrent5_per_sample$GeneSymbol | 
+                  !Gene2B %in% fusion_recurrent5_per_sample$GeneSymbol) %>%
+  unique()
+
+
+
+
+```
+
+
+
+
+
+```{r}
+setwd(outputfolder)
+
+# filter for putative driver genes and mutifused genes per sample
+putative_driver_annotated_fusions <- fusion_calls %>%
+  # dplyr::filter(!Gene1A %in% fusion_recurrent5_per_sample$GeneSymbol |
+  #                 !Gene2A %in% fusion_recurrent5_per_sample$GeneSymbol |
+  #                 !Gene1B %in% fusion_recurrent5_per_sample$GeneSymbol | 
+  #                 !Gene2B %in% fusion_recurrent5_per_sample$GeneSymbol) %>%
+  dplyr::filter(!is.na(Gene1A_anno) | !is.na(Gene1B_anno) | !is.na(Gene2A_anno) | !is.na(Gene2B_anno)) %>%
+  unique()
+  
+# merge putative annotated oncogenic and scavenged back non-oncogenic annotated, recurrent fusions
+putative_driver_fusions<-rbind(QCGeneFiltered_recFusionUniq,putative_driver_annotated_fusions)  
+
+write.table(putative_driver_fusions,"PutativeDriverFusion.tsv",sep="\t",quote=FALSE,row.names = FALSE)
```

Comment:
I would go with
```suggestion
write.table(putative_driver_fusions, file.path(outputfolder, "PutativeDriverFusion.tsv"),sep="\t",quote=FALSE,row.names = FALSE)
```
and remove `setwd(outputfolder)` at line 245

Reply:
ok done!